workflow/rules/peak_utils.smk

```diff
@@ -120,3 +126,82 @@ rule MEME:
         --oc {params.oc}_ame ${{tmp}}/{params.outfa} \\
         {params.meme_euk_db} {params.meme_vertebrates_db} {params.meme_genome_db}
         """
+
+
+pkcaller2homer_size = defaultdict(lambda: "given")
+pkcaller2homer_size.update({
+    'macsNarow': "200", 
+    'macsBroad': "given", 
+    'genrich': "given", 
+    'SEACR': "given"
+})
+
+
+rule HOMER:
+     input:
+         up_file                         = join(
+                                             diffbind_dir,
+                                             "{contrast}-{PeakTool}",
+                                             "{contrast}-{PeakTool}_Diffbind" + block_add + "_{differential_app}_up.bed",
+                                           ),
+         down_file                       = join(
+                                             diffbind_dir,
+                                             "{contrast}-{PeakTool}",
+                                             "{contrast}-{PeakTool}_Diffbind" + block_add + "_{differential_app}_down.bed",
+                                           ),
+     output:
+         down_motifs                     = [join(homer_dir, "DOWN_{contrast}_{PeakTool}_{differential_app}", fn) for fn in homer_output_targets],
+         up_motifs                       = [join(homer_dir, "UP_{contrast}_{PeakTool}_{differential_app}", fn) for fn in homer_output_targets],
+     params:
+         rname                           = 'HOMER',
+         genomefa                        = genome_fasta,
+         genomealias                     = genome,
+         out_dir_up                      = join(homer_dir, "UP_{contrast}_{PeakTool}_{differential_app}"),
+         out_dir_down                    = join(homer_dir, "DOWN_{contrast}_{PeakTool}_{differential_app}"),
+         seq_length                      = "8,10",
+         motif_finding_region            = pkcaller2homer_size["{PeakTool}"],
+         tmpdir                          = tmpdir
+     threads:
+         int(cluster['HOMER'].get('threads', cluster['__default__']['threads']))
+     shell:
+        """
+        if [ ! -d "{params.tmpdir}" ]; then mkdir -p "{params.tmpdir}"; fi
+        tmp=$(mktemp -d -p "{params.tmpdir}")
+        trap 'rm -rf "${{tmp}}"' EXIT
+        export TMPDIR="${{tmp}}" # used by sort
+        module load homer/4.11.1
+        cd ${{TMPDIR}}
+        min() {{
+            printf "%s\\n" "${{@:2}}" | sort "$1" | head -n1
+        }}
+        [ -d "/fdb/homer/genomes/{params.genomealias}" ] || {{ echo "Homer does not support this genome!" >&2; exit 1; }}
```

Comment:
Can we add this path as a key to the genome.json file as a reference file? This will make it more clear that the pipeline relies on that reference file.